runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TransformTranslator.java

```diff
@@ -569,8 +568,8 @@ private static Partitioner getPartitioner(EvaluationContext context) {
     Long bundleSize =
         context.getSerializableOptions().get().as(SparkPipelineOptions.class).getBundleSize();
     return (bundleSize > 0)
-        ? null
-        : new HashPartitioner(context.getSparkContext().defaultParallelism());
+        ? CustomSparkHashPartitioner.of(context.getSparkContext().defaultParallelism())
```

Comment:
This logic seems to be accidentally flipped?

Reply:
agreed, this logic is odd, I don't think `EvalutionContext` is enough to properly resolve paralellism for the shuffle operation

In case `bundleSize > 0`, we should reuse paralellism of the input RDD. This way we would get rid of `nullable` case and we can simplify GBK and CPK translations.

Spark's default partitioner does the same thing: org.apache.spark.Partitioner#defaultPartitioner